environment.go

```diff
@@ -1050,13 +1050,7 @@ func (e *envState) SaveConfig(ctx context.Context) error {
 	// this is a hacky workaround to any plugins that have fields that are maps, since
 	// deserializing these fields from yaml does not maintain the typing information
 	var copy Settings
-	registeredTypes := []interface{}{
-		map[interface{}]interface{}{},
-		map[string]interface{}{},
-		[]interface{}{},
-		[]util.KeyValuePair{},
-	}
-	err := util.DeepCopy(*e.settings, &copy, registeredTypes)
+	err := util.DeepCopy(*e.settings, &copy)
 	if err != nil {
 		return errors.Wrap(err, "copying settings")
 	}
```

Comment:
Is the `gob.Register` on L1064 still needed if we're not using gob encoding anymore? It seems like, aside from the deep copy, all of the extra logic here (from the "hacky workaround" comment above to the special handling of `Plugins` below) isn't needed if we don't have to deal with registering the types specially. We could remove the comment and all the special plugin-handling logic below if they aren't relevant.